rust/common-runtime/benches/runtime_bench.rs

```diff
@@ -110,28 +62,32 @@ fn run_benchmark(c: &mut Criterion) {
         .build()
         .unwrap();
 
-    let (instance_id, runtime_client, _builder_client, _runtime_task, _builder_task) =
-        async_runtime.block_on(async {
-            let (mut runtime_client, mut builder_client, runtime_task, builder_task) =
-                init_servers().await.unwrap();
-            let module_id = build(&mut builder_client).await.unwrap();
-            let instance_id = instantiate(&mut runtime_client, module_id).await.unwrap();
-            (
-                instance_id,
-                runtime_client,
-                builder_client,
-                runtime_task,
-                builder_task,
-            )
-        });
+    let (runtime, instance_id, _builder_task) = async_runtime.block_on(async {
+        let (builder_address, builder_task) = init_build_server().await.unwrap();
+        let mut runtime = Runtime::new().unwrap();
+        let (module, initial_io) =
+            BenchModule::new_basic_js_module().into_components(Some(builder_address.clone()));
+        let instance_id = runtime.compile(module, initial_io).await.unwrap();
+
+        (runtime, instance_id, builder_task)
+    });
+
+    let bench_input = {
+        let input = [("foo".into(), Value::String("updated foo".into()))].into();
+        let output_shape = runtime.output_shape(&instance_id).unwrap().to_owned();
+        RuntimeIo::new(input, output_shape)
+    };
 
-    let input = 0;
     c.bench_with_input(
-        BenchmarkId::new("Runtime::run_module", &input),
-        &input,
-        move |b, _| {
-            b.to_async(&async_runtime)
-                .iter(|| run(runtime_client.clone(), instance_id.clone()));
+        BenchmarkId::new("Runtime::run_module", ""),
+        &bench_input,
+        |b, runtime_io| {
+            b.to_async(&async_runtime).iter(|| {
+                // As RuntimeIo is owned, containing dynamic BTreeMaps,
+                // we can't avoid either construction or cloning, responsible
+                // for ~70us
```

Comment:
We can perhaps make it optional (in which case we'll just keep the previous one around), but in practice every invocation will involve passing in a newly populated `InputOutput`.